api/opentrons/drivers/smoothie_drivers/v3_0_0/driver_3_0.py

```
@@ -35,78 +39,95 @@
           'SET_SPEED': 'G0F',
           'SET_POWER': 'M907'}
 
+homed_positions = {
+    'X': 394, 'Y': 344, 'Z': 227, 'A': 227, 'B': 18.9997, 'C': 18.9997
+}
+
+def _position_at_expected_thresh(position, cached_position):
+    '''Checks parsed position against cache and some threshold'''
+    for axis in AXES.lower():
+        if axis not in position:
+            position[axis] = cached_position[axis]
+        if abs(position[axis] - cached_position[axis]) > POSITION_THRESH:
```

suggest moving threshold to argument and pass it via call. in this case this is minor, but overall helps with testability when the function you are testing checks for something.

the name of the function suggests boolean is returned, but it looks like it mutates first argument being passed, returns True or rasies an exception.

if we want to raise an exception i would suggest writing 

```python
if _position_at_expected_thresh(position, cached_position, thresh):
    self.update_cached_position(position)
else:
    raise ...
```

this way arguments are not mutated and functions do what names suggest

---

removed this functionality based on your other suggestion